components/usage/pkg/controller/reconciler.go

```diff
@@ -76,36 +75,26 @@ func (u *UsageReconciler) Reconcile() (err error) {
 	}
 	log.WithField("usage_reconcile_status", status).Info("Reconcile completed.")
 
-	// For now, write the report to a temp directory so we can manually retrieve it
-	dir := os.TempDir()
-	f, err := ioutil.TempFile(dir, fmt.Sprintf("%s-*", now.Format(time.RFC3339)))
-	if err != nil {
-		return fmt.Errorf("failed to create temporary file: %w", err)
-	}
-	defer f.Close()
-
-	enc := json.NewEncoder(f)
-	err = enc.Encode(report)
+	reportBytes := &bytes.Buffer{}
+	gz := gzip.NewWriter(reportBytes)
+	err = json.NewEncoder(gz).Encode(report)
```

Comment:
This is where the json report is gzipped.

Reply:
I'd recommend moving this into the Storage Client. As it stands it's confusing because the client sets the content type gzipped, but nothing in the API signature indicates it should be gzipped ahead of it. 

Ideally, the Client takes a bytes buffer and wraps it in a gzip writer, then writes the whole thing as a stream

Reply:
Or even can take an `io.Writer` which is a much better interface for wrapping

Reply:
The `UploadFile` method really should only take usage reports - this object storage signed upload URL that it obtains will place the file in the `usage-report` bucket. 

As such, I've renamed the method on the interface and changed its argument to be a report rather than an `io.Reader` (aa9f2d4ca4686a4fef89dd43f1aaaf8a160eecb3)